ntbs-service/Startup.cs

```diff
@@ -534,8 +555,12 @@ private void AddMicrosoftGraphServices(IServiceCollection services, AzureAdOptio
 
 
         // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
-        public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
+        public void Configure(IApplicationBuilder app, IWebHostEnvironment env,
+            TelemetryConfiguration telemetryConfiguration)
         {
+            telemetryConfiguration.DisableTelemetry =
+                !Configuration.GetValue(Constants.ApplicationInsightsEnabled, false);
+
```

Comment:
As mentioned above, this is where `telemetryConfiguration.DisableTelemetry` is set